### PR TITLE
CCPP cmake cleanup. Use ESMF 8.1.1. Update MOM6

### PR DESCRIPTION
**Pull request by DusanJovic-NOAA**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [ ] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [ ] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

Update compiler flags in `cmake/Intel.cmake` and `cmake/GNU.cmke` so that CCPP can use the flags set by UFS instead of maintaining it's own copy of all flags. See https://github.com/NOAA-EMC/fv3atm/pull/292 for details.

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes #<issue_number>
- fixes noaa-emc/fv3atm/issues/<issue_number>

## Testing

Tested oh Hera. All tests are bit reproducible. No baseline changes.

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3

## Dependencies

If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on NOAA-EMC/fv3atm/pull/292

### CODE REVIEW

**Review by climbfuji — APPROVED** [review]
Looks good to me.

File: cmake/GNU.cmake [hunk 1]
```diff
@@ -25,3 +25,9 @@ else()
     set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -O2")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2")
 endif()
+
+# For ccpp
+set(CMAKE_Fortran_FLAGS_DEFAULT_PREC "-fdefault-real-8 -fdefault-double-8")
+set(CMAKE_C_FLAGS_RELEASE       "-O2 -fPIC" CACHE STRING "" FORCE)
+set(CMAKE_CXX_FLAGS_RELEASE     "-O2 -fPIC" CACHE STRING "" FORCE)
```
Comment (aerorahul) [inline cmake/GNU.cmake:32]: There is no C/CXX code in CCPP.
This should be moved out of here or removed as the case may be.

**Review by aerorahul — APPROVED** [review]
looks good.

**Comment by jiandewang** [comment]
look good, MOM6 is pointing to the right repo. at this moment

**Comment by jiandewang** [comment]
@DusanJovic-NOAA can you add "update MOM6" in the PR title ?

**Review by jiandewang — APPROVED** [review]

**Comment by BrianCurtis-NOAA** [comment]
I'm looking on jet: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/625568033/20210507150015/ufs-weather-model/tests/log_jet.intel
I can't see any jobs running on squeue. Looks like it failed and failed without commenting here either.

**Comment by DusanJovic-NOAA** [comment]
```
$ tail /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/625568033/20210507150015/ufs-weather-model/tests/log_jet.intel/run_030_fv3_2threads.log
52003912.0                RUNNING              fv3.exe 
11 min. TEST 030 fv3_2threads is RUNNING,  status: RUNNING jobid 52003912
Slurm unknown status COMPLETI. Check sacct ...
52003912                   FAILED        rt_159547_030 
52003912.ba+               FAILED                batch 
52003912.ex+            COMPLETED               extern 
52003912.0              COMPLETED              fv3.exe 
12 min. TEST 030 fv3_2threads is FAILED,  status: COMPLETI jobid 52003912
rt_utils.sh: line 205: echo: write error: Disk quota exceeded
/mnt/lfs4/HFIP/h-nems/emc.nemspara/autort/pr/625568033/20210507150015/ufs-weather-model/tests/run_test.sh: line 21: echo: write error: Disk quota exceeded
```

**Comment by BrianCurtis-NOAA** [comment]
-->grep -ril "quota" .
./run_030_fv3_2threads.log
./run_019_cpld_restart_bmarkfrac.log

IDK where to look to find who's taking all of our data.

**Comment by jiandewang** [comment]
@DusanJovic-NOAA Let me know when all RT passed, so that I can ask reviewers to approve MOM6 PR and do the merge.

**Comment by DusanJovic-NOAA** [comment]
All regression tests passed. Ready to merge.

**Comment by jiandewang** [comment]
> All regression tests passed. Ready to merge.

@DusanJovic-NOAA  MOM6 has been merged to dev/emc branch, you can switch submodule now

**Review by junwang-noaa — APPROVED** [review]
the code can be committed after the subcomponents point back to the official repoes

**Pull request merged**